src/fiber/fiber.ml

```diff
@@ -498,6 +481,16 @@ module Throttle = struct
 end
 
 let run t =
-  let result = ref None in
-  EC.apply (fun () -> t) () (fun x -> result := Some x);
-  !result
+  EC.new_run (fun () ->
+      let result = ref None in
+      EC.apply (fun () -> t) () (fun x -> result := Some x);
```

Comment:
What abut introducing a helper like:

```
let apply_and_set_result t =
  let result = ref None in
  EC.apply (fun () -> t) () (fun x -> result := Some x);
  result
```

and then using it in both `run` and `run2`.

Reply:
Seems good, doing that